crates/pixi_command_dispatcher/src/install_pixi/mod.rs

```diff
@@ -118,6 +119,34 @@ impl InstallPixiEnvironmentSpec {
         install_reporter: Option<Box<dyn rattler::install::Reporter>>,
     ) -> Result<InstallPixiEnvironmentResult, CommandDispatcherError<InstallPixiEnvironmentError>>
     {
+        // Acquire a lock on the prefix to prevent concurrent installations
+        let guard = AsyncPrefixGuard::new(self.prefix.path())
+            .await
+            .map_err(|e| {
+                CommandDispatcherError::Failed(InstallPixiEnvironmentError::AcquireLock(
+                    self.prefix.clone(),
+                    e,
+                ))
+            })?;
+
+        let mut write_guard = guard.write().await.map_err(|e| {
+            CommandDispatcherError::Failed(InstallPixiEnvironmentError::AcquireLock(
+                self.prefix.clone(),
+                e,
+            ))
+        })?;
+
+        // Mark that we're beginning installation. We always call begin() because:
+        // 1. We have the exclusive write lock, so no one else is installing
+        // 2. If a previous process crashed, the state might be stale
+        // 3. The begin() method is idempotent if already in "Installing" state
+        write_guard.begin().await.map_err(|e| {
```

Comment:
Do you think it makes sense to check for `is_ready` state, and if it was already installed early out?

We are doing the same in `create_exec_prefix`, so I'm thinking maybe we can apply this pattern here?